src/brsTypes/components/RoSGNode.ts

```diff
@@ -730,9 +704,35 @@ export function createNodeByType(interpreter: Interpreter, type: BrsString) {
     let typeDef = interpreter.environment.nodeDefMap.get(type.value);
     if (type.value === "Node") {
         return new RoSGNode([]);
-    } else if (typeDef !== undefined) {
-        //TODO: use typeDef object to tack on all the bells & whistles of a custom node
+    } else if (typeDef) {
```

Comment:
Can you move adding/setting the fields to a different function? I think we can do something like this:
```
export function createNodeByType(interpreter: Interpreter, type: BrsString) {
  let typeDef = interpreter.environment.nodeDefMap.get(type.value);
  let newNode = new RoSGNode([]);
  if (typeDef){
     addFields(newNode)
  }

return newNode;
}
```

Reply:
We will also want to pass `type.value` into the RoSGNode constructor so that custom components have their actual type instead of the default `"Node"`